sources/osgUtil/IntersectFunctor.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+var vec3 = require('osg/glMatrix').vec3;
+var mat4 = require('osg/glMatrix').mat4;
+var PrimitiveIndexFunctor = require('osg/PrimitiveIndexFunctor');
+var intersectionEnums = require('osgUtil/intersectionEnums');
+
+// Settings are needed.
+var IntersectFunctor = function() {
+    this._hit = false;
+
+    // geom info
+    this._primitiveIndex = 0;
+    this._vertices = undefined;
```

Comment:
no preference if it should be reseted or not, reset is mostly used internally to avoid recreating several times the functor. IntersectFunctor are mostly internal usage anyway.
I'll let @cedricpinson see what should be reseted or not.

The advantage is that we don't have to have 2 differents codepaths with intersect function : 
- one that takes indices
- one another that takes vertices
(by the way there were a few bugs concerning that where the behaviour was not the same)

Reply:
I mean the functor keeps a reference to the vertices so it could cause leaks.

Reply:
yep I should call reset at the end of the intersect function instead of the beginning.

Reply:
Good solution, agree